SonarQube.TeamBuild.PreProcessor/SonarWebService.cs

```diff
@@ -123,39 +118,28 @@ public IEnumerable<string> GetActiveRuleKeys(string qualityProfile)
             return result;
         }
 
-        private string GetUrl(string format, params string[] args)
+        // TODO Should be replaced by calls to api/languages/list after min(SQ version) >= 5.1
+        /// <summary>
+        /// Get all keys of all installed plugins
+        /// </summary>
+        public IEnumerable<string> GetInstalledPlugins()
         {
-            var queryString = string.Format(System.Globalization.CultureInfo.InvariantCulture, format, args.Select(a => Uri.EscapeUriString(a)).ToArray());
-            if (!queryString.StartsWith("/", StringComparison.OrdinalIgnoreCase))
-            {
-                queryString = '/' + queryString;
-            }
-            return Server + queryString;
-        }
-
-        #region IDispose implementation
+            var ws = GetUrl("/api/updatecenter/installed_plugins");
+            var contents = Downloader.Download(ws);
 
-        private bool disposed;
+            var plugins = JArray.Parse(contents);
 
-        public void Dispose()
-        {
-            this.Dispose(true);
-            GC.SuppressFinalize(this);
+            return plugins.Select(plugin => plugin["key"].ToString());
```

Comment:
Would be safer to use InvariantCulture in all product ToString() calls that deal with internal values.